### PR TITLE
Larger play thread pool, async fossil client

### LINKED ISSUE #7140: Optimize ThreadPool usage, make more use of async IO
We should take a closer look at the threadpool configuration, compare https://www.playframework.com/documentation/2.8.x/ThreadPools

 - We should probably get rid of the Global Execution Context
 - We should probably increase the number of threads in the play thread pool, because we have some blocking IO
 - We should reduce the amount of blocking IO

What may be blocking
 - talking to postgres (though slick claims async, maybe it has its own thread pool?)
 - talking to redis (there may be alternative clients, such as lettuce or salad (I’m not making these up))
 - talking to dataset storage via https (fixed in #7137)
 - talking to google cloud storage, s3
 - talking to file system
 - talking to fossildb (wip in #7139)

### PR DESCRIPTION
**Pull request by fm3**
- Remove usages of ExecutionContext.global. [Using that discouraged because there are no limits to how many threads it spawns](https://stackoverflow.com/questions/30805337/plays-execution-contexts-vs-scala-global). Before this PR, we randomly used it whereever it was too hard to inject the play execution context.
 - Adds a sensible config for the play execution context. Before this PR, the default config was used, which yielded way too few threads for the amount of blocking operations we have.
 - Adapts most of the fossildb gprc client to async and adapts its error handling accordingly. With one exception, which is getMultipleKeys, because that is used in Iterators, which are harder to rewrite to async (this may be a future optimization)

### TODO
 - [x] get rid of global execution contest
 - [x] adapt akka config for some more threads
   - [x] also in datastore + tracingstore
 - [x] go for more async io usages
   - [x] async fossil client
     - [x] adapt health check
     - [x] adapt get request
     - [x] adapt other requests
     - [x] error handling (exceptions are thrown inside the future?)

### URL of deployed dev instance (used for testing):
- https://threadpools.webknossos.xyz

### Steps to test:
- Test that data loading for remote datasets still works
- Test that annotating (save, reload, download) still works
- To actually test the improvements in parallelism, I used the test route added in [4c96ce4](https://github.com/scalableminds/webknossos/pull/7139/commits/4c96ce4d3ffb04698460cfb8e2c205d03fcd1da2) – sending a lot of parallel requests to /tracings/test showed in the logging how the requests were answered in parallel

### Issues:
- fixes #7140

------
- [x] Removed dev-only changes like prints and application.conf edits
- [x] Considered [common edge cases](../blob/master/.github/common_edge_cases.md)
- [x] Needs datastore update after deployment

### CODE REVIEW

File: util/src/main/scala/com/scalableminds/util/xml/XMLWrites.scala [hunk 1]
```diff
@@ -1,23 +0,0 @@
-package com.scalableminds.util.xml
-
-import javax.xml.stream.XMLStreamWriter
-
-import com.scalableminds.util.tools.Fox
-
-import scala.concurrent.ExecutionContext.Implicits.global
```
Comment (fm3) [inline util/src/main/scala/com/scalableminds/util/xml/XMLWrites.scala:7]: the whole file was unused

File: project/plugins.sbt [hunk 2]
```diff
@@ -20,4 +20,4 @@ addSbtPlugin("com.sksamuel.scapegoat" %% "sbt-scapegoat" % "1.1.1")
 addSbtPlugin("net.vonbuchholtz" % "sbt-dependency-check" % "3.1.3")
 
 //protocol buffers
-libraryDependencies += "com.thesamet.scalapb" %% "compilerplugin" % "0.11.12"
+libraryDependencies += "com.thesamet.scalapb" %% "compilerplugin" % "0.11.13"
```
Comment (fm3) [inline project/plugins.sbt:23]: Upgrade not really needed for this, but shouldn’t hurt (I added it while debugging something which turned out to be caused elsewhere)

**Review by frcroth — APPROVED** [review]
